Basic_viewer/doc/Basic_viewer/CGAL/Graphics_scene_options.h

```diff
@@ -0,0 +1,101 @@
+          typename VolumeDescriptor=void>
+struct Graphics_scene_options
+{
+public:
+  /// `std::function` that returns `true` if the given vertex must be ignored, `false` otherwise.
+  /// Returns `false` by default.
+  std::function<bool(const DS &, VertexDescriptor)> ignore_vertex;
+
+  /// `std::function` that returns `true` if the given edge must be ignored, `false` otherwise.
+  /// Returns `true` by default.
+  std::function<bool(const DS &, EdgeDescriptor)> ignore_edge;
+
+  /// `std::function` that returns `true` if the given face must be ignored, `false` otherwise.
+  /// Returns `true` by default.
+  std::function<bool(const DS &, FaceDescriptor)> ignore_face;
+
+  /// `std::function` that returns `true` if the given volume must be ignored, `false` otherwise.
+  /// Exists only if `VolumeDescriptor` is not `void`.
+  /// Returns `false` by default.
+  std::function<bool(const DS &, VolumeDescriptor)> ignore_volume;
+
+  /// `std::function` that returns `true` if the given vertex is colored, `false` otherwise.
+  /// Returns `false` by default.
+  std::function<bool(const DS &, VertexDescriptor)> colored_vertex;
+
+  /// `std::function` that returns `true` if the given edge is colored, `false` otherwise.
+  /// Returns `false` by default.
+  std::function<bool(const DS &, EdgeDescriptor)> colored_edge;
+
+  /// `std::function` that returns `true` if the given face is colored, `false` otherwise.
+  /// Returns `false` by default.
+  std::function<bool(const DS &, FaceDescriptor)> colored_face;
+
+  /// `std::function` that returns `true` if the given volume is colored, `false` otherwise.
+  /// Returns `false` by default.
+  /// Exists only if `VolumeDescriptor` is not `void`.
+  std::function<bool(const DS &, VolumeDescriptor)> colored_volume;
+
+  /// `std::function` that returns `true` if the given face is in wireframe, `false` otherwise.
+  /// Returns `false` by default.
+  std::function<bool(const DS &, FaceDescriptor)> face_wireframe;
+
+  /// `std::function` that returns `true` if the given volume is in wireframe, `false` otherwise.
+  /// Returns `false` by default.
+  /// Exists only if `VolumeDescriptor` is not `void`.
+  std::function<bool(const DS &, VolumeDescriptor)> volume_wireframe;
+
+  /// `std::function` that returns the color of the given vertex.
+  /// nullptr by default.
+  std::function<CGAL::IO::Color(const DS &, VertexDescriptor)> vertex_color;
+
+  /// `std::function` that returns the color of the given edge.
+  /// nullptr by default.
+  std::function<CGAL::IO::Color(const DS &, EdgeDescriptor)> edge_color;
+
+  /// `std::function` that returns the color of the given face.
+  /// nullptr by default.
+  std::function<CGAL::IO::Color(const DS &, FaceDescriptor)> face_color;
+
+  /// `std::function` that returns the color of the given volume.
+  /// nullptr by default.
+  /// Exists only if `VolumeDescriptor` is not `void`.
+  std::function<CGAL::IO::Color(const DS &, VolumeDescriptor)> volume_color;
```

Comment:
It seems that API does not permit to set the monochrome colors.

Reply:
The use of mono color is is not in the scene option but in the basic viewer.

Reply:
If `colored_face`/`is_face_colored` return `false`, then what is the expected color of the corresponding element?

If a user is completely agnostic of the basic viewer class, and only uses `add_to_graphics_scene` and `draw_graphics_scene`, then that setting is not available. Probably the default color is black, and that is what any user would expect.

But instead of being a property of the viewer, should not it be a property of the graphics scene?